src/Cmd.mli

```diff
@@ -1,19 +1,31 @@
 (** Interface for running system commands *)
 
-type t
+open Import
 
-val make :
-  ?env:string Js.Dict.t -> cmd:Path.t -> unit -> (t, string) result Promise.t
+type shell = string
+
+type spawn =
+  { bin : Path.t
+  ; args : string list
+  }
+
+type t =
+  | Shell of shell
+  | Spawn of spawn
 
 type stdout = string
 
 type stderr = string
 
-val output :
-     args:string Js.Array.t
-  -> ?cwd:Path.t
-  -> ?stdin:string
-  -> t
-  -> (stdout, stderr) result Promise.t
+val append : spawn -> string list -> spawn
+
+val checkSpawn : spawn -> spawn Or_error.t Promise.t
 
-val binPath : t -> Path.t
+val check : t -> t Or_error.t Promise.t
+
+val toSpawn : t -> spawn
+
+val log : ?cwd:string -> ?result:ChildProcess.return -> t -> unit
```

Comment:
I don't think `cwd` or `result` are used anywhere. Shall we just get rid of them?

Reply:
`result` is used here https://github.com/ocamllabs/vscode-ocaml-platform/blob/a180852b1fff330b19447dcdb3d0c70354c7e91a/src/Cmd.ml#L104

`cwd` can be removed